apple/Handlers/RNNativeViewHandler.mm

```diff
@@ -51,7 +61,16 @@ - (void)touchesMoved:(NSSet<RNGHUITouch *> *)touches withEvent:(UIEvent *)event
 - (void)touchesEnded:(NSSet<RNGHUITouch *> *)touches withEvent:(UIEvent *)event
 {
   [_gestureHandler.pointerTracker touchesEnded:touches withEvent:event];
-  self.state = UIGestureRecognizerStateFailed;
+
+#ifdef RCT_NEW_ARCH_ENABLED
+  if ([self.view.superview isKindOfClass:[RCTParagraphComponentView class]]) {
+#else
+  if ([self.view isKindOfClass:[RCTTextView class]]) {
+#endif
```

Comment:
This can be a method

Reply:
Sure, changed in [dd02ac1](https://github.com/software-mansion/react-native-gesture-handler/pull/3202/commits/dd02ac137a4b8830e317074f9ee98755aedab85b)